cryostatio/cryostat
# fix(recording): backwards compatibility for restart parameter

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #1670 

## Description of the change:

Added back the `restart` parameter for backwards compatibility. Future major releases can remove it if needed but should highlight this.

## Motivation for the change:

See #1670 

> I recall we had some discussion around this change before but I can't remember exactly what we landed on. Since this is changing in a minor release version the API "shouldn't" change in incompatible ways, but I'm okay with changing a query parameter like this. Or we could have the endpoint handler handle both forms and prefer the newer one if both are supplied.

## How to manually test:

The old legacy graphQL query with `restart` parameter should work fine:

```graphql
query {
    environmentNodes(filter: { name: "JDP" }) {
        descendantTargets {
            doStartRecording(recording: {
                name: "foo",
                template: "Profiling",
                templateType: "TARGET",
                duration: 30,
                restart: true
                }) {
                    name
                    state
            }
        }
    }
}
```

Also, operations on web client also succeeds.
